instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/__init__.py

```diff
@@ -97,7 +97,7 @@ def _patched_api_call(self, original_func, instance, args, kwargs):
         ) as span:
 
             operation = None
-            if args:
+            if args and span.is_recording():
                 operation = args[0]
                 span.resource = Resource(
```

Comment:
This is a separate issue, but it doesn't seem right to me that you can set a Resource directly on a Span. Seems strange since OTLP only has Resource once for all the spans sent in a request: https://github.com/open-telemetry/opentelemetry-proto/blob/30d237e1ff3ab7aa50e0922b5bebdd93505090af/opentelemetry/proto/trace/v1/trace.proto#L28-L31

I couldn't find it in the spec either.

Reply:
https://github.com/open-telemetry/opentelemetry-python/issues/1180

Reply:
Yeah this seemed strange to me too. Is there a use case for dynamically setting the `Resource` though? How would we even do this? Modify `TracerPovider` 's resources, create span, and then remove resource from `TracerProvider`? Seems kind of unecessary.

Reply:
It also seems that `boto` and `botocore` are the only instrumentations that do this, and in turn, have a dependency on the sdk. We can continue the discussion in the issue.